wire-ios/Wire-iOS/Sources/UserInterface/Location/LocationSelectionViewController.swift

```diff
@@ -44,26 +62,33 @@ final class LocationSelectionViewController: UIViewController {
     let locationButtonContainer = UIView()
     var sendControllerHeightConstraint: NSLayoutConstraint?
 
-    fileprivate var mapView = MKMapView()
-    fileprivate let toolBar = ModalTopBar()
-    fileprivate let locationManager = CLLocationManager()
-    fileprivate let geocoder = CLGeocoder()
-    fileprivate let sendViewController = LocationSendViewController()
-    fileprivate let pointAnnotation = MKPointAnnotation()
+    private var mapViewController = MapViewController()
+    private let toolBar = ModalTopBar()
+    private let locationManager = CLLocationManager()
+    private let geocoder = CLGeocoder()
+    private let sendViewController = LocationSendViewController()
+    private let pointAnnotation = MKPointAnnotation()
     private lazy var annotationView: MKPinAnnotationView = MKPinAnnotationView(annotation: pointAnnotation, reuseIdentifier: String(describing: type(of: self)))
-    fileprivate var userShowedInitially = false
-    fileprivate var mapDidRender = false
+    private var userShowedInitially = false
+    private var mapDidRender = false
+
+    lazy var appLocationManager: AppLocationManager = {
+        let manager = AppLocationManager(locationManager: CLLocationManager())
+        manager.delegate = self
+        return manager
+    }()
 
-    fileprivate var userLocationAuthorized: Bool {
-        let status = locationManager.authorizationStatus
+    private var userLocationAuthorized: Bool {
+        let status = appLocationManager.authorizationStatus
         return status == .authorizedAlways || status == .authorizedWhenInUse
     }
```

Comment:
Should this implementation `userLocationAuthorized` be moved to the manager?

Reply:
Done: https://github.com/wireapp/wire-ios/commit/8d186be1f4bab13dfe60eff797bd7693ec9cfa30